### ISSUE TITLE
Empty wavelength array in MultiSlit cal products

### ISSUE THREAD

**Issue by hbushouse** [STScI-JWST/jwst#1711]
When calwebb_spec2 is run on modes that are not NIRSpec fixed-slit or MSA (such as NIRCam grism) the output _cal product, which comes from a MultiSlitModel, includes wavelength extensions that have an empty array. For example, a NIRCam grism exposure produces:
```
In [2]: fits.info('jw11111001001_01104_00001_nrcalong_cal.fits')
Filename: jw11111001001_01104_00001_nrcalong_cal.fits
No.    Name      Ver    Type      Cards   Dimensions   Format
  0  PRIMARY       1 PrimaryHDU     340   ()      
  1  SCI           1 ImageHDU        24   (585, 81)   float32   
  2  DQ            1 ImageHDU        11   (585, 81)   int32 (rescales to uint32)   
  3  ERR           1 ImageHDU        10   (585, 81)   float32   
  4  RELSENS       1 BinTableHDU     14   0R x 2C   [D, D]   
  5  WAVELENGTH    1 ImageHDU         9   (0, 0)      
  6  SCI           2 ImageHDU        24   (435, 81)   float32   
  7  DQ            2 ImageHDU        11   (435, 81)   int32 (rescales to uint32)   
  8  ERR           2 ImageHDU        10   (435, 81)   float32   
  9  RELSENS       2 BinTableHDU     14   0R x 2C   [D, D]   
 10  WAVELENGTH    2 ImageHDU         9   (0, 0)      
 11  ASDF          1 ImageHDU         7   (35889,)   uint8   
```
The wavelength extension is probably appearing just because it's defined in the MultiSlitModel, but it's empty, because modes like grism don't populate it. It would be nice to find a way to not have the wavelength extension appear at all when it's not used.

**Comment by hbushouse** [comment]
See https://jira.stsci.edu/browse/JP-217

**Comment by nden** [comment]
We need to check  if this is still the case with the current master where the `wavelength` array in the slitdata schema does not have default value.

**Comment by hbushouse** [comment]
As far as I know the test I ran about an hour ago that generated the results shown above were using the latest master.

**Comment by hbushouse** [comment]
The data files for the test I ran are sitting in /grp/jwst/ssb/test_build7.1/examples_for_dms/level2/nrc_grism/test/ if you want to try it for yourself. So by the way this suggests that grism processing IS working using the SlitModel updates.

**Comment by hbushouse** [comment]
@nden is right (as usual): with the latest updates to datamodels, schema validation, etc. the empty wavelength arrays no longer show up in the _cal products from WFSS exposures:
```
In [8]: fits.info('jw11111001001_01104_00001_nrcalong_cal.fits')
Filename: jw11111001001_01104_00001_nrcalong_cal.fits
No.    Name      Ver    Type      Cards   Dimensions   Format
  0  PRIMARY       1 PrimaryHDU     341   ()      
  1  SCI           1 ImageHDU        21   (545, 0)      
  2  DQ            1 ImageHDU        11   (545, 0)      
  3  ERR           1 ImageHDU         9   (545, 0)      
  4  RELSENS       1 BinTableHDU     14   21R x 2C   [D, D]   
  5  SCI           2 ImageHDU        21   (353, 0)      
  6  DQ            2 ImageHDU        11   (353, 0)      
  7  ERR           2 ImageHDU         9   (353, 0)      
  8  RELSENS       2 BinTableHDU     14   21R x 2C   [D, D]   
  9  ASDF          1 NonstandardExtHDU      8   ()      
```

**Comment by hbushouse** [comment]
Fixed by #1070, #1700, #1709, and #1734.
